table/block_based_table_reader.cc

```diff
@@ -153,8 +153,9 @@ class BlockBasedTable::IndexReader {
   virtual ~IndexReader() {}
 
   // Create an iterator for index access.
-  // An iter is passed in, if it is not null, update this one and return it
-  // If it is null, create a new Iterator
+  // If a non-null iter is passed in it MIGHT be used instead of creating a new
```

Comment:
It looks messy to have to check if result is on heap or stack. Can we update `PartitionIndexReader` to use the input `iter`?